File: app/src/main/java/com/jyotirmayg/mytestproject/feature/login/LoginViewModel.java
package com.jyotirmayg.mytestproject.feature.login;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.jyotirmayg.mytestproject.data.LoginStatus;

/**
 * @author jyoti
 * @created on 17-05-2022
 */
public class LoginViewModel extends ViewModel {

    private final MutableLiveData<LoginStatus> _loginLiveData = new MutableLiveData();
    LiveData<LoginStatus> loginStatus = _loginLiveData;

    /**
     * This is called from the layout file to initiate login.
     *
     * @param userName holds the username entered by the user.
     * @param password holds the password entered by the user.
     */
    void onClickLogin(String userName, String password) {
        LoginStatus loginStatus;
        Boolean status = doLogin(userName, password);
        if (status) {
            loginStatus = new LoginStatus(true, "Login successful");
        } else {
            loginStatus = new LoginStatus(false, "Login failed");
        }
        _loginLiveData.postValue(loginStatus);
    }

    Boolean doLogin(String userName, String password) {
        return userName.equals("admin") && password.equals("admin123");
    }
}

File: app/src/main/java/com/jyotirmayg/mytestproject/view/login/LoginViewModel.kt
package com.jyotirmayg.mytestproject.view.login

import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.jyotirmayg.mytestproject.R
import com.jyotirmayg.mytestproject.base.Application
import com.jyotirmayg.mytestproject.data.LoginStatus

/**
 * @author jyoti
 * @created on 14-05-2022
 */
class LoginViewModel(private val app: Application) : AndroidViewModel(app) {

    private val _loginLiveData = MutableLiveData<LoginStatus>()
    val loginStatus: LiveData<LoginStatus> = _loginLiveData

    /**
     * This is called from the layout file to initiate login.
     * @param userName holds the username entered by the user.
     * @param password holds the password entered by the user.
     * */
    fun onClickLogin(userName: String, password: String) {
        val loginStatus: LoginStatus = if (userName == "admin" && password == "admin123") {
            LoginStatus(true, app.getString(R.string.login_success))
        } else {
            LoginStatus(false, app.getString(R.string.login_failed))
        }
        _loginLiveData.postValue(loginStatus)
    }
}